deepchem/data/datasets.py

```diff
@@ -156,43 +248,83 @@ def get_task_names(self):
 
   @property
   def X(self):
-    """Get the X vector for this dataset as a single numpy array."""
+    """Get the X vector for this dataset as a single numpy array.
+
+    Returns
+    -------
+    Numpy array of features `X`.
+    """
     raise NotImplementedError()
 
   @property
   def y(self):
-    """Get the y vector for this dataset as a single numpy array."""
+    """Get the y vector for this dataset as a single numpy array.
+
+    Returns
+    -------
+    Numpy array of labels `y`.
+    """
     raise NotImplementedError()
 
   @property
   def ids(self):
-    """Get the ids vector for this dataset as a single numpy array."""
+    """Get the ids vector for this dataset as a single numpy array.
+
+    Returns
+    -------
+    Numpy array of identifiers `ids`.
+    """
 
     raise NotImplementedError()
 
   @property
   def w(self):
-    """Get the weight vector for this dataset as a single numpy array."""
+    """Get the weight vector for this dataset as a single numpy array.
+
+    Returns
+    -------
+    Numpy array of weights `w`.
+    """
     raise NotImplementedError()
 
+  def __repr__(self):
+    """Convert self to REPL print representation."""
+    threshold = dc.utils.get_print_threshold()
+    id_str = np.array2string(self.ids, threshold=threshold)
```

Comment:
This is potentially very slow.  For a DiskDataset, `ids` calls `itershards()` to iterate over the entire dataset and load in the ids (and everything else).

Reply:
Good catch. This is a little tricky to handle well. I ended up adding a new flag to utilities `max_print_size` that controls how big an array can get before the printing of `self.ids` is turned off.